File: src/main/java/kitchenpos/menu/tobe/domain/repository/MenuGroupRepository.java
package kitchenpos.menu.tobe.domain.repository;

import java.util.List;
import java.util.Optional;
import java.util.UUID;
import kitchenpos.menu.tobe.domain.entity.MenuGroup;

public interface MenuGroupRepository {

    MenuGroup save(final MenuGroup menuGroup);

    Optional<MenuGroup> findById(final UUID id);

    List<MenuGroup> findAll();
}

File: src/main/kotlin/kitchenpos/menu/tobe/domain/MenuGroupRepository.kt
package kitchenpos.menu.tobe.domain

import java.util.*
import org.springframework.context.annotation.Primary
import org.springframework.stereotype.Repository

@Repository("tobeMenuGroupRepository")
@Primary
interface MenuGroupRepository {
    fun save(menuGroup: MenuGroup): MenuGroup

    fun findById(id: UUID): Optional<MenuGroup>

    fun findAll(): List<MenuGroup>
}
